### DISCUSSION TITLE
Automated string resource unit tests

### DISCUSSION THREAD

**Discussion by FloEdelmann** [streetcomplete/StreetComplete#4059] [Ideas]
Some tasks regarding the string resources in various languages could be checked (and maybe even fixed) automatically:

* [#4032 (comment)](https://github.com/streetcomplete/StreetComplete/issues/4032#issuecomment-1128174791): `…` instead of `...` in `en`, `en-rGB` and `de` (maybe more)
* [#4032 (comment)](https://github.com/streetcomplete/StreetComplete/issues/4032#issuecomment-1128174791): `’` instead of `'` in `en`, `en-rGB`, `fr` and `de` (maybe more)
* [#4032](https://github.com/streetcomplete/StreetComplete/issues/4032): no space before `…` in `en` and `en-rGB`
* https://de.wikipedia.org/wiki/Auslassungspunkte#Deutsch: non-breaking space before `…` in `de`
* thin non-breaking space between `z. B.` in `de`
* https://github.com/streetcomplete/StreetComplete/commit/6aca0dc9476034466d04949ffe870075932edb96: `e.g.` instead of `f.e.` in `en` and `en-rGB`
* …

Does it make sense to write an automated test for these tasks? Where should that test go; `test`, `androidTest` or `buildSrc`?

**Comment by matkoniecz** [comment]
From what I remember build process already checks whether `%s` and similar are preserved in translated strings.

I expect that other similar tasks should go into the same place?

**Comment by westnordost** [comment]
`Code -> Inspect Code...` already has some of it. However, finding these is one thing, applying the fixes is the more time consuming task, as it would have to be done in POEditor. And then, you might as well do this search on POEditor.
